manual/library/builtin.etex

```diff
@@ -190,7 +192,10 @@ exception Failure of string
 \index{Failure@\verb`Failure`}
 \begin{ocamldocdescription}
    Exception raised by library functions to signal that they are
-   undefined on the given arguments.
+   undefined on the given arguments.  The string gives some information
+   to the programmer; it is \emph{bad practice} to pattern match
```

Comment:
> About the fact that it should be frowned upon, we can hardly say that while the standard library makes extensive use of it.

We can say that because the standard library was written 20 years ago and practices have changed now. I still see a lot of people designing APIs around the `Failure` exception for error handling which is very bad.

> Moreover, result is impractical without the use of appropriate monadic connectors — which again are not provided by the standard library.

What's wrong with pattern matching on `result` ? We have been using options for ages without having the `option` monad in the stdlib.